### PR TITLE
MINIFICPP-1731 - Linter targets should not be built by default on Win…

### PR DESCRIPTION
**Pull request by adam-markovics**
…dows

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -872,25 +872,27 @@ include(BuildDocs)
...
-		COMMAND python3 ${CMAKE_SOURCE_DIR}/thirdparty/google-styleguide/run_linter.py -q -i
-		${linted_dir}
-	)
-	math(EXPR linted_dir_counter "${linted_dir_counter}+1")
-endforeach()
-
-# Main linter target that depends on every other
-add_custom_target(linter DEPENDS ${root_linter_target_names} ${extensions})
+if (ENABLE_LINTER)
+    get_property(extensions GLOBAL PROPERTY EXTENSION-LINTERS)
```
Comment (lordgamez) [inline CMakeLists.txt:876]: This cmake file is indented with tabs so the new content should also use tabs instead of spaces

File: CMakeLists.txt [hunk 2]
```diff
@@ -872,25 +872,27 @@ include(BuildDocs)
...
-	add_custom_target("${linter_target_name}"
-		COMMAND python3 ${CMAKE_SOURCE_DIR}/thirdparty/google-styleguide/run_linter.py -q -i
-		${linted_dir}
-	)
-	math(EXPR linted_dir_counter "${linted_dir_counter}+1")
-endforeach()
-
-# Main linter target that depends on every other
-add_custom_target(linter DEPENDS ${root_linter_target_names} ${extensions})
+if (ENABLE_LINTER)
```
Comment (lordgamez) [inline CMakeLists.txt:875]: Could you please explain how does this change the behavior on Windows? It looks like the ENABLE_LINTER is still ON by default and we did not change anything that is specific to Windows.

Comment (szaszm) [reply CMakeLists.txt:875]: The root-linter-* targets used to run on Windows even if ENABLE_LINTER was OFF. Now we don't even create the targets if ENABLE_LINTER=OFF.

Comment (lordgamez) [reply CMakeLists.txt:875]: Oh now I see, so it's more about the creation of the target than about building the target. It's clear now, thanks!

File: CMakeLists.txt (hunk 1 above)
Comment (lordgamez) [reply CMakeLists.txt]: Actually as I checked although as the file already has 15 other lines where spaces are used for indentation it is not that important.

Comment (adam-markovics) [reply CMakeLists.txt]: Anyways, I made an update.

**Review by lordgamez — APPROVED** [review]

**Review by adamdebreceni — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

**Pull request closed without merging**